os/arch/arm/src: Fix Ameba incorrect sem_wait condition

ziyik opened the pull request:
- Correct sem_wait while loop check condition
- Standardize sem_wait return value check

sunghan-chang reviewed (commented):
Could you split commits for each chipset?
1. rtl8721csm and amebad
2. rtl8720e and amebalite
3. rtl8730e and amebasmart

Verification and product team want to get necessary change only. Not all.

ziyik commented:
> Could you split commits for each chipset?
> 
> 1. rtl8721csm and amebad
> 2. rtl8720e and amebalite
> 3. rtl8730e and amebasmart
> 
> Verification and product team want to get necessary change only. Not all.

Done, I have split it into 3 commits

sunghan-chang commented:
@ziyik Is this verified (tested)? Could you let me know what you tested?

ziyik commented:
> @ziyik Is this verified (tested)? Could you let me know what you tested?

Tested build, boot up with init.
The detail test can only be done on Thursday, the peripheral pic is on leave today.

sunghan-chang commented:
@ziyik Because almost are at end of dev, let's merge this after test.

ziyik commented:
> @ziyik Because almost are at end of dev, let's merge this after test.

Yes sure, we will run some test and update you the result once it's done.

zhongnuo-tang commented:
Hi @sunghan-chang,
This PR includes three parts
1. Standardized Return Value: Replaced hardcoded return value 0 with OK for sem_wait (where OK is defined as 0).
2. Condition Check Update: Modified the condition check for the return value of sem_wait from <0 to != OK. As we verified, the return value of sem_wait is either OK (success) or a negative value (failure), so this change does not alter the logic.
3. Logic Correction: Corrected the while condition in AmebaD's I2C implementation from while (errno == EINTR) to while (ret != OK), as the previous condition was logically incorrect.

In my understanding:
Points 1 and 2 do not require testing, as they do not introduce any functional or logical changes.
For point 3, testing is necessary to validate the fix. However, we currently do not have access to the TP1x module required for verification.
Could you share your thoughts for this?

sunghan-chang commented:
> Hi @sunghan-chang, This PR includes three parts
> 
> 1. Standardized Return Value: Replaced hardcoded return value 0 with OK for sem_wait (where OK is defined as 0).
> 2. Condition Check Update: Modified the condition check for the return value of sem_wait from <0 to != OK. As we verified, the return value of sem_wait is either OK (success) or a negative value (failure), so this change does not alter the logic.
> 3. Logic Correction: Corrected the while condition in AmebaD's I2C implementation from while (errno == EINTR) to while (ret != OK), as the previous condition was logically incorrect.
> 
> In my understanding: Points 1 and 2 do not require testing, as they do not introduce any functional or logical changes. For point 3, testing is necessary to validate the fix. However, we currently do not have access to the TP1x module required for verification. Could you share your thoughts for this?

@ziyik @zhongnuo-tang Agree. Then, could you clean up the commits like below?
1. Let's say code clean up for amebasmart, 0 -> OK.
2. In Amebad and Amebalite, let's split commits for 0 -> OK and return value checking. 0 -> OK is just code clean up, changing return value checking is `fix incorrect sem_wait condition`.

sunghan-chang commented:
@ziyik @zhongnuo-tang Verification team will check the commits and will ask what it is, test result and affection of the system. We should show that it's just code clean up for amebasmart.

ziyik commented:
> > Hi @sunghan-chang, This PR includes three parts
> > 
> > 1. Standardized Return Value: Replaced hardcoded return value 0 with OK for sem_wait (where OK is defined as 0).
> > 2. Condition Check Update: Modified the condition check for the return value of sem_wait from <0 to != OK. As we verified, the return value of sem_wait is either OK (success) or a negative value (failure), so this change does not alter the logic.
> > 3. Logic Correction: Corrected the while condition in AmebaD's I2C implementation from while (errno == EINTR) to while (ret != OK), as the previous condition was logically incorrect.
> > 
> > In my understanding: Points 1 and 2 do not require testing, as they do not introduce any functional or logical changes. For point 3, testing is necessary to validate the fix. However, we currently do not have access to the TP1x module required for verification. Could you share your thoughts for this?
> 
> @ziyik @zhongnuo-tang Agree. Then, could you clean up the commits like below?
> 
> 1. Let's say code clean up for amebasmart, 0 -> OK.
> 2. In Amebad and Amebalite, let's split commits for 0 -> OK and return value checking. 0 -> OK is just code clean up, changing return value checking is `fix incorrect sem_wait condition`.

I have split the commits, is the commit titles okay?
